aws/retry/retryable_error.go

```diff
@@ -95,8 +95,13 @@ func (r RetryableConnectionError) IsErrorRetryable(err error) aws.Ternary {
 	var timeoutErr interface{ Timeout() bool }
 	var urlErr *url.Error
 	var netOpErr *net.OpError
+	var dnsError *net.DNSError
 
 	switch {
+	case errors.As(err, &dnsError):
+		// NXDOMAIN errors should not be retried
+		retryable = false
```

Comment:
**correctness**: Not all instances of `net.DNSError` are related to NXDOMAIN errors. This should be predicated on the fields in [net.DNSError](https://pkg.go.dev/net#DNSError).

e.g. something like

```go
case errors.As(err, &dnsError):
    // NXDOMAIN and other non temporary DNS errors should not be retried
    retryable = !dnsError.IsNotFound && dnsError.IsTemporary
```